datarobot_batch_scoring/reader.py

```diff
@@ -584,6 +586,11 @@ def auto_sampler(dataset, encoding, ui):
              'recommended lines per sample: {}'.format(csv_lines, avg_line,
                                                        lines_per_sample))
     ui.info('auto_sampler: total time seconds - {}'.format(time() - t0))
+    if max_prediction_explanations:
+        # The prediction explanation scores 10 extra rows for each column for each row.
+        # But not more than for 30 columns. So the slowdown cannot be more than 300x.
```

Comment:
slowdown of what? I don't get this comment.

Reply:
@doleks this code is not supposed to limit number of lines per batch based on runtime, it just limits batch size to avoid frontend level reject of payload

2 MB payload on some hardcore blender can hit server-side timeout easily, but there is no code in batch-scoring to avoid this. If you want to implement something like this, we need to start with small batches and scale based on actual prediction time, passing suggested batch size via shared memory variable from `network` or `writer` to `reader`

Reply:
Bollocks! I thought it is some time-based heuristic.

Instead of implementing the retry mechanism, it probably would be better to use scoring benchmarks that the core engine does. But in any case, it is too elaborate for what I'm trying to do. So I will remove my change.

Reply:
I didn't mean to retry failed requests, but instead start with small batches and gradually increase batch size to meet some predefined prediction time like 10 seconds per batch

Reply:
You are right. But measuring time is not deterministic either. It is prone to fluctuations based on the load. And because your proposal is self-adjusting, you need to think about all possible scenarios. Like many scripts being fired at once.
Not today.